core/infrastructure/secret.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource/compare"
+	"github.com/kiegroup/kogito-operator/api"
+	"github.com/kiegroup/kogito-operator/core/client/kubernetes"
+	"github.com/kiegroup/kogito-operator/core/operator"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"path"
+)
+
+// SecretHandler ...
+type SecretHandler interface {
+	FetchSecret(key types.NamespacedName) (*corev1.Secret, error)
+	MustFetchSecret(key types.NamespacedName) (*corev1.Secret, error)
+	GetComparator() compare.MapComparator
+	MountAsVolume(deployment *appsv1.Deployment, volumeReference api.VolumeReferenceInterface) error
+	MountAsEnvFrom(deployment *appsv1.Deployment, cmName string)
+}
+
+type secretHandler struct {
+	operator.Context
+}
+
+// NewSecretHandler ...
+func NewSecretHandler(context operator.Context) SecretHandler {
+	return &secretHandler{
+		Context: context,
+	}
+}
+
+func (c *secretHandler) FetchSecret(key types.NamespacedName) (*corev1.Secret, error) {
+	c.Log.Debug("fetching secret.")
+	secret := &corev1.Secret{}
+	if exists, err := kubernetes.ResourceC(c.Client).FetchWithKey(key, secret); err != nil {
+		return nil, err
+	} else if !exists {
+		c.Log.Debug("Secret not found.")
+		return nil, nil
+	} else {
+		c.Log.Debug("Successfully fetch deployed Secret")
+		return secret, nil
+	}
+}
+
+func (c *secretHandler) MustFetchSecret(key types.NamespacedName) (*corev1.Secret, error) {
+	c.Log.Debug("going to must fetch deployed secret.")
+	if secret, err := c.FetchSecret(key); err != nil {
+		return nil, err
+	} else if secret == nil {
+		return nil, fmt.Errorf("Secret not found with name %s in namespace %s ", key.Name, key.Namespace)
+	} else {
+		return secret, nil
+	}
+}
+
+func (c *secretHandler) MountAsVolume(deployment *appsv1.Deployment, volumeReference api.VolumeReferenceInterface) error {
+	if err := c.appendVolumeIntoDeployment(deployment, volumeReference); err != nil {
+		return err
+	}
+	if err := c.appendVolumeMountIntoDeployment(deployment, volumeReference); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *secretHandler) appendVolumeIntoDeployment(deployment *appsv1.Deployment, secretReference api.VolumeReferenceInterface) error {
```

Comment:
Nitpick - the returned error value is always nil, can be removed

Reply:
done